### PR TITLE
Add TFDS preprocessors

### PR DESCRIPTION
**Pull request by kurarrr**
## Motivation and Context

The context is https://github.com/blue-oil/blueoil/issues/727

And this includes PR https://github.com/blue-oil/blueoil/pull/728

## Description

I added tfds preprocessors and augmentors which treat tensor without breaking changes.
In config file, you can use them by

```
DATASET.PRE_PROCESSOR = None
DATASET.TFDS_PRE_PROCESSOR = Sequence([
    TFResize(size=IMAGE_SIZE),
    TFPerImageStandardization()
])
DATASET.AUGMENTOR = None
DATASET.TFDS_AUGMENTOR = Sequence([
    TFFlipLeftRight()
])
```

instead of
```
DATASET.PRE_PROCESSOR =  Sequence([
    Resize(size=IMAGE_SIZE),
    PerImageStandardization()
])
DATASET.AUGMENTOR = Sequence([
    FlipLeftRight()
])
```

Note that not all preprocessors are covered and take same arguments as the existing ones.
And you can use arbitrary ones by defining callable object in config file.

## How has this been tested?

Also updated tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lmnet/lmnet/tfds_pre_processor.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    """Standardization per image.
+    """
+
+    def __call__(self, image, **kwargs):
+        return tf.image.per_image_standardization(image)
+
+class TFResizeWithGtBoxes(Processor):
+    """Resize image with gt boxes.
+
+    Use :func:`~resize_with_gt_boxes` inside.
```
Comment (tfujiwar) [inline lmnet/lmnet/tfds_pre_processor.py]: I think this is not correct and not necessary. Please remove this.

File: lmnet/lmnet/tfds_pre_processor.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        gt_boxes = gt_boxes[:, 3].assign(gt_boxes[:, 3] * scale[0])
+
+        gt_boxes = gt_boxes[:, 0].assign(tf.minimum(
+                gt_boxes[:, 0], width - gt_boxes[:, 2]))
+        gt_boxes = gt_boxes[:, 1].assign(tf.minimum(
+                gt_boxes[:, 1], width - gt_boxes[:, 3]))
+
+class TFResize(Processor):
+    """Resize an image
+    """
```
Comment (tfujiwar) [inline lmnet/lmnet/tfds_pre_processor.py]: Please write this docstring in one line.
https://www.python.org/dev/peps/pep-0008/#documentation-strings

File: lmnet/lmnet/tfds_pre_processor.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    def __call__(self, image, **kwargs):
+        """
+        Args:
+            image (tf.Tensor): an image tensor sized (orig_height, orig_width, channel)
+        """
+        return tf.image.resize(image, self.size)
+
+class TFPerImageStandardization(Processor):
+    """Standardization per image.
+    """
```
Comment (tfujiwar) [inline lmnet/lmnet/tfds_pre_processor.py]: Please write this docstring in one line.
https://www.python.org/dev/peps/pep-0008/#documentation-strings

File: lmnet/lmnet/tfds_pre_processor.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import tensorflow as tf
+from lmnet.data_processor import Processor
+
+def tf_resize_with_gt_boxes(image, gt_boxes, size=(256,256)):
+    """Resize an image and gt_boxes.
+
+    Args:
+        image (np.ndarray): An image numpy array.
+        gt_boxes (np.ndarray): Ground truth boxes in the image. shape is [num_boxes, 5(x, y, width, height)].
```
Comment (tfujiwar) [inline lmnet/lmnet/tfds_pre_processor.py]: Maybe `tf.Tensor` is correct.

File: lmnet/lmnet/tfds_augmentor.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        return dict({'image': image }, **kwargs)
+
+
+
+def _random_flip_left_right_bounding_box(image, gt_boxes, seed):
+    """Flip left right only bounding box.
+
+    Args:
+        image    (tensor): image
+        gt_boxes (tensor): bounding boxes. shape is [num_boxes, 5(x, y, w, h, class_id)]
```
Comment (tfujiwar) [inline lmnet/lmnet/tfds_augmentor.py]: Maybe `tf.Tensor` is better.

**Comment by kurarrr** [comment]
@tfujiwar 
Thank you for the review!
I'll check later.

**Pull request closed without merging**